src/EnergyPlus/CurveManager.cc

```diff
@@ -358,33 +352,35 @@ namespace CurveManager {
-            state.dataCurveManager->PerfCurve(CurveNum).curveType = CurveType::BiQuadratic;
-            state.dataCurveManager->PerfCurve(CurveNum).ObjectType = CurrentModuleObject;
-            state.dataCurveManager->PerfCurve(CurveNum).NumDims = 2;
-            state.dataCurveManager->PerfCurve(CurveNum).InterpolationType = InterpType::EvaluateCurveToLimits;
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff1 = Numbers(1);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff2 = Numbers(2);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff3 = Numbers(3);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff4 = Numbers(4);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff5 = Numbers(5);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff6 = Numbers(6);
-            state.dataCurveManager->PerfCurve(CurveNum).Var1Min = Numbers(7);
-            state.dataCurveManager->PerfCurve(CurveNum).Var1Max = Numbers(8);
-            state.dataCurveManager->PerfCurve(CurveNum).Var2Min = Numbers(9);
-            state.dataCurveManager->PerfCurve(CurveNum).Var2Max = Numbers(10);
+            thisCurve.Name = Alphas(1);
+            thisCurve.curveType = CurveType::BiQuadratic;
+            thisCurve.ObjectType = CurrentModuleObject;
+            thisCurve.NumDims = 2;
+            thisCurve.InterpolationType = InterpType::EvaluateCurveToLimits;
+            thisCurve.coeff[0] = Numbers(1);
+            thisCurve.coeff[1] = Numbers(2);
+            thisCurve.coeff[2] = Numbers(3);
+            thisCurve.coeff[3] = Numbers(4);
+            thisCurve.coeff[4] = Numbers(5);
+            thisCurve.coeff[5] = Numbers(6);
+            thisCurve.inputLimits[0].min = Numbers(7);
+            thisCurve.inputLimits[0].max = Numbers(8);
+            thisCurve.inputLimits[1].min = Numbers(9);
+            thisCurve.inputLimits[1].max = Numbers(10);
             if (NumNumbers > 10 && !state.dataIPShortCut->lNumericFieldBlanks(11)) {
-                state.dataCurveManager->PerfCurve(CurveNum).CurveMin = Numbers(11);
-                state.dataCurveManager->PerfCurve(CurveNum).CurveMinPresent = true;
+                thisCurve.outputLimits.min = Numbers(11);
+                thisCurve.outputLimits.minPresent = true;
             }
             if (NumNumbers > 11 && !state.dataIPShortCut->lNumericFieldBlanks(12)) {
-                state.dataCurveManager->PerfCurve(CurveNum).CurveMax = Numbers(12);
-                state.dataCurveManager->PerfCurve(CurveNum).CurveMaxPresent = true;
+                thisCurve.outputLimits.max = Numbers(12);
+                thisCurve.outputLimits.maxPresent = true;
             }
 
             if (Numbers(7) > Numbers(8)) { // error
-                ShowSevereError(state, "GetCurveInput: For " + CurrentModuleObject + ": " + Alphas(1));
+                ShowSevereError(state, format("GetCurveInput: For {}: ", CurrentModuleObject));
```

Comment:
Can probably encapsulate these into a `ShowCurveNumericalRelationError`.

Reply:
Yep, there will be some common functionality extracted for sure.